feature/s3/manager/download.go

```diff
@@ -401,6 +404,7 @@ func (d *downloader) tryDownloadChunk(params *s3.GetObjectInput, w io.Writer) (i
 		return 0, err
 	}
 	d.setTotalBytes(resp) // Set total if not yet set.
+	d.setETag(resp)
```

Comment:
Shouldn't this be set only once instead of at each chunk?

Reply:
yea, the func would only set it in first GET when etag is empty

Reply:
I see, this is because of the mutex we added to setETag. Unfortunately this means all threads will have to wait for the lock since we use the same mutex as the downloader and they all try to call setETag when getting a result. Can we use something like [sync.OnceValue](https://pkg.go.dev/sync#OnceValue) to set this instead, so we don't have all go routines waiting on the lock?